src/git.rs

```diff
@@ -8,7 +8,7 @@ use crate::semver::SemVer;
 /// git helper for common operations
 pub(crate) struct GitHelper {
     pub(crate) repo: Repository,
-    version_map: HashMap<Oid, VersionAndTag>,
+    version_map: HashMap<Oid, Vec<VersionAndTag>>,
```

Comment:
I tried to fix by keeping only the highest version on a sha, but this looks a better solution.

```diff
@@ -147,14 +147,16 @@ fn make_oid_version_map(repo: &Repository, prefix: &str) -> HashMap<Oid, Version
     for tag in tags.iter().flatten().filter(|tag| tag.starts_with(prefix)) {
         if let Ok(oid) = repo.revparse_single(tag).map(object_to_target_commit_id) {
             if let Ok(version) = Version::parse(tag.trim_start_matches(prefix)) {
-                map.insert(
-                    oid,
-                    VersionAndTag {
-                        tag: tag.to_owned(),
-                        version: SemVer(version),
-                        commit_sha: oid.to_string(),
-                    },
-                );
+                let version_and_tag = VersionAndTag {
+                    tag: tag.to_owned(),
+                    version: SemVer(version.clone()),
+                    commit_sha: oid.to_string(),
+                };
+                if let Some(old) = map.insert(oid, version_and_tag) {
+                    if old.version.0 > version {
+                        map.insert(oid, old);
+                    }
+                };
             }
         }
     }
```